docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -1,15 +1,135 @@
+    </div>
+  );
+}
+```
+
+And now test your route by starting your Front-Commerce server and navigate to
+[http://localhost:4000/faq](http://localhost:4000/faq).
+
+You should now see your brand new page.
+
+## Create an Emitter class
+
+Let's create a file in our project, called `FAQPageReached.ts`:
+
+```tsx title="app/server-events/events/FAQPageReached.ts"
+import { ServerEvent } from "@front-commerce/core/events";
+
+// Name of the event that will be dispatched to the external services
+const FAQPageReachedName = "FAQPageReachedEvent";
+
+// Object that will be sent to external services
+type FAQPageReachedEventType = {
+  question: string;
+};
+
+export default class FAQPageReachedEvent extends ServerEvent<FAQPageReachedEventType> {
+  static event_type = FAQPageReachedName;
+
+  // In our guide we will check if customer pass
+  // And we will submit a metadata object containing metadatas about the event
+  // Here we will retrieve the locale
+  constructor(question: string, locale: string) {
+    // Call the constructor of the base class, the new Date() match the event creation date
+    super(FAQPageReachedName, new Date(), { question }, { locale });
+  }
+}
+```
+
+That's it!
+
+## Use the Emitter
+
+Now that we've created an Emitter, let's use it in our page:
+
+```tsx title="app/routes/_main.faq.tsx"
+import { FrontCommerceApp } from "@front-commerce/remix";
+import FAQPageReachedEvent from "../server-events/events/FAQPageReachedEvent";
+import type { LoaderFunctionArgs } from "@remix-run/node";
+
+//highlight-start
+export async function loader({ context, request }: LoaderFunctionArgs) {
+  // We create a Front-Commerce App to access the ServerEventBus Service
+  const app = new FrontCommerceApp(context.frontCommerce);
+  // We parse the question from the URL
+  const question = new URL(request.url).searchParams.get("question");
+
+  // We emit the event!
+  app.services.ServerEventBus.publish(
+    new FAQPageReachedEvent(String(question), app.config.public.shop.locale)
+  );
+  return {};
+}
+//highlight-end
+
+export default function FAQ() {
+  return (
+    <div>
+      <h1>FAQ</h1>
+    </div>
+  );
+}
+```
+
+## Test our emitter
+
+Now that everything is ready, let's test it! To do so, run the Front-Commerce
+Worker by running:
+
+```shell
+pnpm worker
+```
+
+The worker will boot:
+
+```
+2024-02-20 18:14:07 [info]:     Font-Commerce Server Events Worker
+2024-02-20 18:14:07 [info]:     Please wait booting...
+2024-02-20 18:14:09 [info]:     Worker started
+```
+
+And now let's head to our FAQ page!
+
+You will see new output to the worker:
+
+```
+[ConsoleIntegration]: (1708449459736-0) {"event_type":"FAQPageReachedEvent","created_at":"2024-02-20T17:17:39.735Z","payload":"{\"question\":\"null\"}","metadata":"{\"locale\":\"en-GB\"}"}
+```
+
+Now let's add some question in URL and head to :
+[http://localhost:4000/faq?question=how+to+create+an+event+emitter](http://localhost:4000/faq?question=how+to+create+an+event+emitter)
```

Comment:
If it's added to the faq-demo, then this can be removed.

Reply:
removed